src/sdk/main/include/NodeCreateTransaction.h

```diff
@@ -0,0 +1,326 @@
+   *
+   * @param transactionBody The TransactionBody protobuf object from which to construct.
+   * @throws std::invalid_argument If the input TransactionBody does not represent a NodeCreate transaction.
+   */
+  explicit NodeCreateTransaction(const proto::TransactionBody& transactionBody);
+
+  /**
+   * Construct from a map of TransactionIds to node account IDs and their respective Transaction protobuf objects.
+   *
+   * @param transactions The map of TransactionIds to node account IDs and their respective Transaction protobuf
+   *                     objects.
+   */
+  explicit NodeCreateTransaction(const std::map<TransactionId, std::map<AccountId, proto::Transaction>>& transactions);
+
+  /**
+   * Get the node account identifier.
+   *
+   * @return An account identifier.
+   */
+  [[nodiscard]] const AccountId& getAccountId() const { return mAccountId; };
+
+  /**
+   * Get the short description of the node.
+   *
+   * @return An optional string containing the description.
+   */
+  [[nodiscard]] const std::optional<std::string>& getDescription() const { return mDescription; };
+
+  /**
+   * Get the list of service endpoints for gossip.
+   *
+   * @return A vector of gossip service endpoints.
+   */
+  [[nodiscard]] const std::vector<Endpoint>& getGossipEndpoints() const { return mGossipEndpoints; };
+
+  /**
+   * Get the list of service endpoints for gRPC calls.
+   *
+   * @return A vector of gRPC service endpoints.
+   */
+  [[nodiscard]] const std::vector<Endpoint>& getServiceEndpoints() const { return mServiceEndpoints; };
+
+  /**
+   * Get the certificate used to sign gossip events.
+   *
+   * @return A vector of bytes representing the gossip CA certificate in DER encoding.
+   */
+  [[nodiscard]] const std::vector<std::byte>& getGossipCaCertificate() const { return mGossipCaCertificate; };
+
+  /**
+   * Get the hash of the node gRPC TLS certificate.
+   *
+   * @return A vector of bytes representing the SHA-384 hash of the gRPC TLS certificate.
+   */
+  [[nodiscard]] const std::optional<std::vector<std::byte>>& getGrpcCertificateHash() const
+  {
+    return mGrpcCertificateHash;
+  };
+
+  /**
+   * Get the administrative key controlled by the node operator.
+   *
+   * @return A shared pointer to the administrative Key.
+   */
+  [[nodiscard]] const std::shared_ptr<Key>& getAdminKey() const { return mAdminKey; };
+
+  /**
+   * Set the node account identifier.
+   *
+   * @param accountId The AccountId to set.
+   * @return A reference to this NodeCreateTransaction with the newly-set account identifier.
+   */
+  NodeCreateTransaction& setAccountId(const AccountId& accountId);
+
+  /**
+   * Set the description for the node.
+   *
+   * @param description A string containing the description.
+   * @return A reference to this NodeCreateTransaction with the newly-set description.
+   */
+  NodeCreateTransaction& setDescription(const std::optional<std::string>& description);
```

Comment:
Not sure we'd want to input an `optional` here.. if the user doesn't want to set a description they can just not set one.